packages/core/src/compiler/Style.ts

```diff
@@ -2649,17 +2783,28 @@ const eval1D = (
     case "Vector": {
       return ok(vectorV(elems));
     }
+    case "CollectionAccess": {
+      return ok(vectorV(elems));
+    }
   }
 };
 
 const eval2D = (
-  coll: List<C> | Vector<C>,
-  first: VectorV<ad.Num>,
+  coll: List<C> | Vector<C> | CollectionAccess<C>,
+  first: VectorV<ad.Num> | ListV<ad.Num> | TupV<ad.Num>,
   rest: ArgVal<ad.Num>[]
-): Result<LListV<ad.Num> | MatrixV<ad.Num>, StyleDiagnostics> => {
+): Result<
+  LListV<ad.Num> | MatrixV<ad.Num> | PtListV<ad.Num>,
+  StyleDiagnostics
+> => {
   const elems = [first.contents];
   for (const v of rest) {
-    if (v.tag === "Val" && v.contents.tag === "VectorV") {
+    if (
```

Comment:
Can you explain why we are evaluating 2D lists and tuples now?

Reply:
I'm guessing this has something to do with the usage of `collectIntoVal` later. How does this impact evaluation of regular values? `eval2D` is also called by `evalListOrVector`, for instance.